controllers/ocsinitialization/ocsoperator_config.go

```diff
@@ -0,0 +1,142 @@
+package ocsinitialization
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	// This configmap is purely for the OCS operator to use
+	ocsOperatorConfigName = "ocs-operator-config"
+)
+
+// ensureOCSOperatorConfig ensures that the OCS operator config is present in the
+// cluster & has the correct values
+func (r *OCSInitializationReconciler) ensureOCSOperatorConfig(initialData *ocsv1.OCSInitialization) error {
+	ocsOperatorConfig := &corev1.ConfigMap{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: ocsOperatorConfigName, Namespace: initialData.Namespace}, ocsOperatorConfig)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// If it does not exist, create the ConfigMap
+			return r.Client.Create(context.TODO(), newOCSOperatorConfig(initialData.Namespace))
+		}
+		return err
+	}
+	resourceVersion := ocsOperatorConfig.ResourceVersion
+
+	//Adding Ownerreference to the configmap if it's not present
+	ownerRefs := ocsOperatorConfig.GetOwnerReferences()
+	if len(ownerRefs) == 0 {
+		err = controllerutil.SetOwnerReference(initialData, ocsOperatorConfig, r.Client.Scheme())
+		if err != nil {
+			return err
+		}
+		err = r.Client.Update(context.TODO(), ocsOperatorConfig)
```

Comment:
I see no need to do this as a separate `Update()` call... just use `CreateOrUpdate()` and pass all the required mutations (ownerRef, clusterId update) there.

Reply:
Addressed now